test/datasets/amazonreviewpolarity_test.py

```diff
@@ -0,0 +1,82 @@
+import os.path
+import random
+import string
+import tarfile
+from collections import defaultdict
+
+from parameterized import parameterized
+from torchtext.datasets.amazonreviewpolarity import AmazonReviewPolarity
+
+from ..common.case_utils import TempDirMixin
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "AmazonReviewPolarity")
+    compressed_dataset_path = os.path.join(
+        base_dir, "amazon_review_polarity_csv.tar.gz"
+    )
+    uncompressed_dataset_dir = os.path.join(base_dir, "amazon_review_polarity_csv")
+    os.makedirs(uncompressed_dataset_dir, exist_ok=True)
+
+    # create empty tar file to skip dataset download
```

Comment:
I think we probably want to add non-empty tar-file with all the content structure as in original compressed file and avoid creating uncompressed directory. This way we would get code-coverage for extraction parts of the datapipes.

Reply:
Just updated the tar-file to contain the contents of the dataset!